cmd/kube-scheduler/app/config/config.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	apiserver "k8s.io/apiserver/pkg/server"
+	"k8s.io/client-go/informers"
+	coreinformers "k8s.io/client-go/informers/core/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	v1core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/kubernetes/cmd/controller-manager/app"
+	"k8s.io/kubernetes/pkg/apis/componentconfig"
+)
+
+// Config has all the context to run a Scheduler
+type Config struct {
+	// config is the scheduler server's configuration object.
+	ComponentConfig componentconfig.KubeSchedulerConfiguration
+
+	InsecureServing        *app.InsecureServingInfo // nil will disable serving on an insecure port
+	InsecureMetricsServing *app.InsecureServingInfo // non-nil if metrics should be served independently
+	Authentication         apiserver.AuthenticationInfo
+	Authorization          apiserver.AuthorizationInfo
+	SecureServing          *apiserver.SecureServingInfo
+
+	SchedulerName                  string
+	Client                         clientset.Interface
+	InformerFactory                informers.SharedInformerFactory
+	PodInformer                    coreinformers.PodInformer
+	AlgorithmSource                componentconfig.SchedulerAlgorithmSource
+	HardPodAffinitySymmetricWeight int32
```

Comment:
Wondering why SchedulerName, AlgorithmSource and HardPodAffinitySymmetricWeight are in this list again when they are already part of ComponentConfig above?

Reply:
I am thinking, either the struct Config could have componentconfig.KubeSchedulerConfiguration and not duplicate its field, or just duplicate fields whatever are needed in the struct, and not include componentconfig.KubeSchedulerConfiguration?